### PR TITLE
R 4.0.0

### PR DESCRIPTION
**Pull request by CJ-Wright**

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Don't merge until we actually have R 4

File: recipe/migrations/r400.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    1
+  # This determines the increment to the build number when the 
+  # migration runs. 
+  # Change this to zero if the new pin increases the number of builds
+  build_number:
+    1
+
+# The name of the feedstock you wish to migrate with dashes replaced by
+# underscores
+r-base:
```
Comment (beckermr) [inline recipe/migrations/r400.yaml:22]: This needs to be r_base per the comment in the file?

Comment (CJ-Wright) [reply recipe/migrations/r400.yaml:22]: And I was the one who wrote that comment! :man_facepalming:

Comment (CJ-Wright) [inline recipe/migrations/r400.yaml:22]:
```suggestion
r_base:
```

This is what happens when bots take over you forget how to do the thing!/s

**Comment by xhochy** [comment]
Is there a reason to wait on pushing the button here?

Probably want approval from @conda-forge/r

**Comment by dpryan79** [comment]
Let's do it!

**Comment by beckermr** [comment]
The bot teams meets today at 12 PM central. We had one more request from @isuruf to put in some code to clean up the R recipes before the migration. Now is a good time to do that since we are rebuilding anyways, but I am worried we will be fighting issues from both the cleanup code and R 4 which might be a lot. We need to conclude on this request before we can push the button.

**Comment by beckermr** [comment]
I think we also want to figure out who can be around to babysit things when as it gets started.

**Comment by dpryan79** [comment]
Is there any update on this? I recall that someone mentioned that the team that decides this was meeting yesterday or today.

**Comment by beckermr** [comment]
We had one hiccup on windows about making sure the v4 packages will work with binaries for R on windows built by CRAN. We don't have a system to test this. If anyone you know does, I think that is the last road block.

**Comment by dpryan79** [comment]
I have a windows laptop I can test that on if it'll help. This is checking if the normal CRAN binary packages work with the conda R, yes?

**Comment by xhochy** [comment]
Yes, please check with the binaries from CRAN whether dplyr and some package that uses fortran works together with our R build.

**Comment by dpryan79** [comment]
Will do, any suggestions on packages using fortran? I don't have any that come to mind.

**Comment by beckermr** [comment]
@mingwandroid @isuruf any suggestions here on R packages that use a lot of fortran here?

Maybe r-glmnet?

**Comment by beckermr** [comment]
I def see fortran source in. r-glmnet, so it is not the worst choice but maybe not the best.

**Comment by dpryan79** [comment]
Both dplyr and glmnet install and R can load them at least. I can test dplyr a bit I suppose since I'm familiar with it. Can't say I ever used glmnet.

```
> sessionInfo()
R version 4.0.0 (2020-04-24)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 18362)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252
[2] LC_CTYPE=English_United States.1252
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C
[5] LC_TIME=English_United States.1252

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] glmnet_3.0-2  Matrix_1.2-18 dplyr_0.8.5

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6     codetools_0.2-16 lattice_0.20-41  fansi_0.4.1
 [5] foreach_1.5.0    crayon_1.3.4     assertthat_0.2.1 grid_4.0.0
 [9] R6_2.4.1         lifecycle_0.2.0  magrittr_1.5     pillar_1.4.3
[13] rlang_0.4.5      cli_2.0.2        vctrs_0.2.4      ellipsis_0.3.0
[17] iterators_1.0.12 tools_4.0.0      glue_1.4.0       purrr_0.3.4
[21] compiler_4.0.0   pkgconfig_2.0.3  shape_1.4.4      tidyselect_1.0.0
[25] tibble_3.0.0
```

**Comment by beckermr** [comment]
For glmnet I’d just make data from a linear model and fit it. I don’t have test code handu

**Comment by dpryan79** [comment]
A simple `left_join()` with dplyr works correctly and the some simple regressions with `glmnet()` seem to have worked properly too.

**Comment by beckermr** [comment]
@xhochy @isuruf are we happy with this test? Seems like things function though for sure we might not be hitting hard edge cases.

**Comment by xhochy** [comment]
👍  Sounds fine from my side.

**Comment by bgruening** [comment]
:tada: :+1:

**Comment by beckermr** [comment]
@bgruening @dpryan79 I am seeing a few R packages with duplicate noarch: generic keys in them? Any ideas on this? This is killing version updates from the bot since it is not valid yaml.

**Comment by dpryan79** [comment]
@beckermr Probably someone messed that up when they made the feedstock.

**Comment by dbast** [comment]
afaik duplicated `noarch: generic` keys comes from an old bot run when noarch was enabled via bot and that was also added to recipes that already contained it.

**Comment by CJ-Wright** [comment]
Lets see how this does. (I might bump the number of PRs later)
